### ISSUE TITLE
expected `i8`, found `u8`

### ISSUE THREAD

**Issue by justinmoon** [raftario/tracing-android#4]
I tried running the example code in my project

```
    tracing_subscriber::registry()
        .with(paranoid_android::layer(
            "com.example.foo",
        ))
        .init();
```

I get the following error:

```
error[E0308]: mismatched types
   --> /Users/justin/.cargo/registry/src/github.com-1ecc6299db9ec823/paranoid-android-0.2.0/src/writer.rs:107:73
    |
107 |                 unsafe { __android_log_buf_write(buffer, priority, tag, message) };
    |                                                                         ^^^^^^^ expected `i8`, found `u8`
    |
    = note: expected raw pointer `*const i8`
               found raw pointer `*const u8`

error[E0308]: mismatched types
   --> /Users/justin/.cargo/registry/src/github.com-1ecc6299db9ec823/paranoid-android-0.2.0/src/writer.rs:209:9
    |
204 |       fn as_ptr(&mut self) -> Option<*const u8> {
    |                               ----------------- expected `Option<*const u8>` because of return type
...
209 | /         CStr::from_bytes_with_nul(self.buf.as_ref())
210 | |             .ok()
211 | |             .map(CStr::as_ptr)
    | |______________________________^ expected `u8`, found `i8`
    |
    = note: expected enum `Option<*const u8>`
               found enum `Option<*const i8>`
```

**Comment by justinmoon** [comment]
I'm building the library on an m1 mac. I think the issue is that `CStr::as_ptr` return `*const c_char` and `c_char` is `i8` on an m1 mac using rust 1.59.0 as seen here in the standard library https://github.com/rust-lang/rust/blob/1.59.0/library/std/src/os/raw/mod.rs#L48.

**Comment by justinmoon** [comment]
Actually no I think this only happens when the target is `i686-linux-android`. Has nothing to do with host machine. But works fine with `aarch64-linux-android` and `armv7-linux-androideabi`.

**Comment by raftario** [comment]
Fixed by version `0.2.1` on crates.io
